adr/0006-reviewer-responsibilies.adoc

```diff
@@ -0,0 +1,74 @@
+
+== Decision Drivers
+
+* Increase clarity and awareness on contributor’s responsibility on reviews and code push
+* Describe the intent and not the current exact state (as tooling is preventing us from having the best solution)
+* Being a perfect solution is _not_ a goal, _better_ than today is.
+* Define ways to measure and improve the time to review.
+
+== Scenarios (optional)
+
+List of scenarios, each described using 2-3 sentences, to capture where the ADR has an impact.
+
+== Considered options
+
+The main alternative is the status quo which roughly is: Guillaume knowing what we are intending to do and some persons talking with Guillaume regularly being sufficiently aware.
+
+== Decision
+
+There are three levels of reviewing.
+
+. Any one can review a Pull Request (PR) and give constructive feedback on different areas (code, documentation, feature experience, security, etc.).
+  This is an optional but recommended step.
+  More eyes means higher quality.
+. A PR is reviewed by a _binding reviewer_ who is fluent in the area being touched: this person is responsible for signing with blood of the usefulness, quality of the code and soundness of the solution.
+  That’s the review of the expert and the most critical one for quality.
+  This review is mandatory.
+.. Unfortunately, we do not always have (designed or effective) experts in all areas so it’s a constant work in progress and might make this step optional in some situations.
+. A PR is reviewed by a committer (push rights) whose responsibilities are:
+.. Do a general review (often from a non expert eye).
+.. Ensure consistency and quality across Quarkus’s codebase (code, concepts, API consistency, documentation).
+.. Detect potential security concerns with an attacker mindset (though not being an actual security professional).
+.. Ensure consistency at the platform level (duplicated or overlapping features, usage integration issues across features etc).
+.. Effectively push the code to the main branch.
+
+=== TODO Open questions to resolve before closing
+
+How is a person moving from casual reviewer to binding reviewer?
+People will tend to request review from binding reviewers leading to a progression problem where casual reviewers will never have the knowledge to be “promoted”.
+We have a few people with no specific area of ownership, that’s specifically acute there.
+
+How can a non binding review see the value of its work considering someone else has to go behind them anyways.
+Feels like double work where the team is very busy.
+
+How can we define SLO for the PR review to help with the _flow_, satisfy contributors while keeping quality high.
+And how to use it as a mechanism to improve and know where to get help?
+The faster the flow, the high an organisation measure in effectiveness, happiness etc.
+
+==== Proposition 1
+
+Define and measure (GitHub APIs) SLO for the time to push or drop / withdraw.
```

Comment:
How would you define that magic number? As explained above, raising awareness is okay (and we have the lottery for this), but taking action is NOT. 

And maybe this issue does not really exist? Do you have examples of PRs that took months to merge? It is not rare for a few weeks to be required because Quarkus has a complex (and pretty large) code base. Some PR have been discussed for multiple months, but that's a different issue.

Reply:
Yes look above, your comment made me do some GitHub queries to try and get some numbers around staleness.